internal/helm/template.go

```diff
@@ -0,0 +1,87 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package helm
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	greenhousev1alpha1 "github.com/cloudoperators/greenhouse/api/v1alpha1"
+)
+
+// ResolveTemplatedValues processes PluginOptionValues with templates and resolves them to concrete values.
+func ResolveTemplatedValues(ctx context.Context, c client.Client, plugin *greenhousev1alpha1.Plugin, optionValues []greenhousev1alpha1.PluginOptionValue) ([]greenhousev1alpha1.PluginOptionValue, error) {
+	resolvedValues := make([]greenhousev1alpha1.PluginOptionValue, 0, len(optionValues))
+
+	for _, optionValue := range optionValues {
+		if optionValue.Template != nil {
+			templateData, err := buildTemplateData(ctx, c, plugin)
+			if err != nil {
+				return nil, fmt.Errorf("failed to build template data: %w", err)
+			}
```

Comment:
The result of buildTemplateData does not change between iterations. So computing it once should be sufficient.